pkg/controller/elasticsearch/stackmon/sidecar.go

```diff
@@ -35,20 +36,37 @@ func Metricbeat(ctx context.Context, client k8s.Client, es esv1.Elasticsearch) (
 	if err != nil {
 		return stackmon.BeatSidecar{}, err
 	}
-	metricbeat, err := stackmon.NewMetricBeatSidecar(
-		ctx,
-		client,
-		commonv1.KbMonitoringAssociationType,
-		&es,
-		es.Spec.Version,
-		metricbeatConfigTemplate,
-		esv1.ESNamer,
-		fmt.Sprintf("%s://localhost:%d", es.Spec.HTTP.Protocol(), network.HTTPPort),
-		"",
-		username,
-		password,
-		es.Spec.HTTP.TLS.Enabled(),
-	)
+
+	v, err := version.Parse(es.Spec.Version)
+	if err != nil {
+		return stackmon.BeatSidecar{}, err
+	}
+
+	caVolume, err := stackmon.CAVolume(client, k8s.ExtractNamespacedName(&es), esv1.ESNamer, commonv1.KbMonitoringAssociationType /*???*/, es.Spec.HTTP.TLS.Enabled())
```

Comment:
I was just doing a first read of your PR and noticed the `/*???*/`, I agree that `KbMonitoringAssociationType` feels a bit odd here but did not dig enough to be sure.

Reply:
I wanted to investigate this myself too and forgot about it. It just has an effect on the name of the volume mount. Changing it would mean a restart for the cluster. But maybe we should do it nonetheless